### PR TITLE
Conformance tests for DeterministicMLPPolicy

### PR DESCRIPTION
**Pull request by nish21**
Add tests to verify `DeterministicMLPPolicyWithModel` works exactly like `DeterministicMLPPolicy`. 

Fix some styling issues that flake/pylint insisted on fixing

In the other PR I was working from a fork and Travis had problems with Docker. Trying to see if this fixes the problem.

### CODE REVIEW

File: garage/tf/policies/deterministic_mlp_policy_with_model.py [hunk 1]
```diff
@@ -96,9 +96,12 @@ def get_action_sym(self, obs_var, name=None, **kwargs):
         """Return action sym according to obs_var."""
         with tf.variable_scope(self._variable_scope):
             action = self.model.build(obs_var, name=name)
-            action = tf.reshape(action, self.action_space.shape)
             return action
 
+    def get_params(self, trainable=True):
+        """Get the trainable variables."""
+        return self.get_trainable_vars()
+
```
Comment (ahtsan) [inline garage/tf/policies/deterministic_mlp_policy_with_model.py]: I think it's time to move `get_params` into base class. Please also update all other policies that have `get_params`.

File: tests/garage/tf/policies/test_deterministic_mlp_policy_with_model.py [hunk 2]
```diff
@@ -69,8 +69,10 @@ def test_get_action_sym(self, obs_dim, action_dim):
 
         action = self.sess.run(
             action_sym, feed_dict={state_input: [obs.flatten()]})
-        assert env.action_space.contains(action)
+        action = np.reshape(action, action_dim)
```
Comment (ahtsan) [inline tests/garage/tf/policies/test_deterministic_mlp_policy_with_model.py]: Use `policy.action_space.unflatten(action)` instead.

File: tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+"""
+Unit test for DeterministicMLPPolicyWithModel.
+
+This test consists of four different DeterministicMLPPolicy: P1, P2, P3
+and P4. P1 and P2 are from DeterministicMLPPolicy, which does not use
+garage.tf.models.DeterministicMLPModel while P3 and P4 do use.
```
Comment (ahtsan) [inline tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py]: garage.tf.models.MLPModel

File: tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            env_spec=self.box_env, hidden_sizes=(32, 32), name='P1')
+        self.policy2 = DeterministicMLPPolicy(
+            env_spec=self.box_env, hidden_sizes=(64, 64), name='P2')
+        self.policy3 = DeterministicMLPPolicyWithModel(
+            env_spec=self.box_env, hidden_sizes=(32, 32), name='P3')
+        self.policy4 = DeterministicMLPPolicyWithModel(
+            env_spec=self.box_env, hidden_sizes=(64, 64), name='P4')
+
+        self.sess.run(tf.global_variables_initializer())
+        for a, b in zip(self.policy3.get_params(), self.policy1.get_params()):
```
Comment (ryanjulian) [inline tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py]: i think you can use Variable.load() for this.

File: tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+from tests.fixtures import TfGraphTestCase
+from tests.fixtures.envs.dummy import DummyBoxEnv
+
+
+class TestDeterministicMLPPolicyWithModelTransit(TfGraphTestCase):
+    @mock.patch('tensorflow.random.normal')
+    def setUp(self, mock_rand):
+        mock_rand.return_value = 0.5
+        super().setUp()
+        self.box_env = TfEnv(DummyBoxEnv())
```
Comment (ryanjulian) [inline tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py]: make sure this doesn't always produce zero-observations (or your tests aren't testing anything)

**Comment by ahtsan** [comment]
Looks good. Please submit after rebasing with latest master locally (Don't press the `Update branch` button above).

**Pull request merged**